lib/matplotlib/dates.py

```diff
@@ -212,47 +212,31 @@ def _to_ordinalf(dt):
     days, preserving hours, minutes, seconds and microseconds.  Return value
     is a :func:`float`.
     """
-
-    if hasattr(dt, 'tzinfo') and dt.tzinfo is not None:
-        delta = dt.tzinfo.utcoffset(dt)
-        if delta is not None:
-            dt -= delta
+    # Convert to UTC
+    tzi = getattr(dt, 'tzinfo', None)
+    if tzi is not None:
+        dt = dt.astimezone(UTC)
```

Comment:
`date` also doesn't have `tzinfo`, so the `astimezone` line will never fire.

On May 2, 2016 4:24:22 PM EDT, Eric Firing <notifications@github.com> wrote:
>> @@ -212,47 +212,31 @@ def _to_ordinalf(dt):
>>      days, preserving hours, minutes, seconds and microseconds. 
>Return value
>>      is a :func:`float`.
>>      """
>> -
>> -    if hasattr(dt, 'tzinfo') and dt.tzinfo is not None:
>> -        delta = dt.tzinfo.utcoffset(dt)
>> -        if delta is not None:
>> -            dt -= delta
>> +    # Convert to UTC
>> +    tzi = getattr(dt, 'tzinfo', None)
>> +    if tzi is not None:
>> +        dt = dt.astimezone(UTC)
>
>I don't think Python 2.7 `datetime.date` has the `astimezone` method.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/matplotlib/matplotlib/pull/6262/files/3e6e8d1324e4aac4c0cf08dcffab451e1ee5f525#r61797587